include/TimePos.h

```diff
@@ -69,42 +71,72 @@ class LMMS_EXPORT TimePos
 	TimePos( const tick_t ticks = 0 );
 
 	TimePos quantize(float) const;
-	TimePos toAbsoluteBar() const;
+	TimePos toAbsoluteBar() const { return getBar() * s_ticksPerBar; }
 
-	TimePos& operator+=( const TimePos& time );
-	TimePos& operator-=( const TimePos& time );
+	TimePos& operator+=(const TimePos& time)
+	{
+		m_ticks += time.m_ticks;
+		return *this;
+	}
+
+	TimePos& operator-=(const TimePos& time)
+	{
+		m_ticks -= time.m_ticks;
+		return *this;
+	}
 
 	// return the bar, rounded down and 0-based
-	bar_t getBar() const;
+	bar_t getBar() const { return m_ticks / s_ticksPerBar; }
+
 	// return the bar, rounded up and 0-based
-	bar_t nextFullBar() const;
+	bar_t nextFullBar() const { return (m_ticks + (s_ticksPerBar - 1)) / s_ticksPerBar; }
+
+	void setTicks(tick_t ticks) { m_ticks = ticks; }
+	tick_t getTicks() const { return m_ticks; }
 
-	void setTicks( tick_t ticks );
-	tick_t getTicks() const;
+	operator int() const { return m_ticks; }
```

Comment:
do we really need this one overload?

Reply:
The overload is mainly for convenience, and I rather keep it to avoid unnecessary changes elsewhere in the codebase that use it currently.